build-system/tasks/check-types.js

```diff
@@ -108,7 +108,10 @@ const TYPE_CHECK_TARGETS = {
   'src-context': ['src/context/**/*.js', ...CORE_SRCS_GLOBS],
   'src-core': CORE_SRCS_GLOBS,
   'src-examiner': ['src/examiner/**/*.js'],
-  'src-experiments': ['src/experiments/**/*.js', ...CORE_SRCS_GLOBS],
+  'src-experiments': {
+    srcGlobs: ['src/experiments/**/*.js', ...CORE_SRCS_GLOBS],
+    externGlobs: ['build-system/externs/amp.extern.js'],
```

Comment:
I think @rcebulko's intention is to not include build-system (BS) externs in any of these folder.

cc @rcebulko

Reply:
I'll add Ryan as reviewer on the un-revert PR after #34796

Reply:
That is correct 👍🏻

Reply:
Related: https://github.com/ampproject/amphtml/pull/34797